lesson18/src/js/main.js

```diff
@@ -0,0 +1,273 @@
+    element === 'previous' && (button.disabled = 'true');
+    fragment.appendChild(button);
+  });
+
+  const slideArea = document.getElementById('js-mainvisual-images');
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagination(item) {
+  const pagination = createElementWithClassName("div", "slider__pagination");
+  const fragment = document.createDocumentFragment();
+  const current = document.createElement("span");
+  current.id = "js-current";
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createElementWithClassName("span", "separation");
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = "js-total";
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.getElementById("js-mainvisual-textarea");
+  textArea.appendChild(pagination).appendChild(fragment);
+}
+
+function renderPaginationBullet(item){
+  const paginationArea = createElementWithClassName("div","mainvisual__pagination__bullet");
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const paginationBullet = createElementWithClassName("span", "bar");
+    paginationBullet.style.width = "calc(width / item)";
+    paginationBullet.setAttribute ("role", "button");
+    paginationBullet.dataset.paginationIndex = i + 1;
+    fragment.appendChild(paginationBullet);
+
+    i === 0 && paginationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideImageArea = document.getElementById("js-mainvisual-images-inner");
+  slideImageArea.appendChild(paginationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.getElementById("js-mainvisual-images-inner");
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function changeActivePagination(currentIndex) {
+  const paginationCurrent = document.getElementById("js-current");
+  paginationCurrent.textContent = currentIndex;
+
+}
+
+function changeActiveImage(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function changeActivePaginationBullet(target){
+  const paginationBullet = document.querySelector('[aria-current="true"');
+  paginationBullet.setAttribute('aria-current', false);
+  const pagiNations = document.querySelector(`[data-pagination-index="${target}"]`);
+  pagiNations.setAttribute("aria-current", true);
+
+}
+
+function clickPaginationBullet(){
+  const paginationBullet = [...document.getElementsByClassName("bar")];
```

Comment:
Bulletって初めて聞きました。どういう意味でしたか?

```suggestion
  const paginationBullets = [...document.getElementsByClassName("bar")];
```

Reply:
Bullet主な意味は弾丸という意味ですが、
他の意味としてリストの黒丸の意味を持っているそうです。
https://eow.alc.co.jp/search?q=bullet

命名に悩みSwiperがこちらを使っていたので、
参考にしました！

Reply:
> 命名に悩みSwiperがこちらを使っていたので、

そうなのですね!